Landing page

enze-l opened the pull request:
closes #8

enze-l committed:
Create first draft of landing page

enze-l committed:
Add landing.css and landing.pug omitted in the last commit

enze-l committed:
Add some formating and text

enze-l committed:
Make tiling grid and added gradiants to features

enze-l committed:
Fixe text breaking boundaries and add styling

- Text in tiles no longer breaks boundaries
- Added buttons to tiles
- Gave tiles own colours

enze-l committed:
Add tick icon

- modified README to reflect IP
- added tick-icon to landing page-buttons

jneidel commented on public/pug/landing.pug:
```diff
@@ -0,0 +1,45 @@
+                    You want to have seperate lists for work, home and shopping
+                    or a project is around the corner? Sort your task to be able 
+                    to focus on what matters for you.
+                a( href="/welcome" class="btn" )
```
Where are these supposed to go?

A sub side describing that part of the functionality?

jneidel commented on public/pug/landing.pug:
```diff
@@ -0,0 +1,45 @@
+        h1  Be more productive
+        p.
+            "What to do" is an easy to use to-do application, designed for
+            individual or colaborativ use.
```
collaborative

jneidel commented:
Background should stay like this? Seems a bit drab as it is right now.

jneidel commented on routes/index.js:
```diff
@@ -1,6 +1,7 @@
 const router = express.Router();
 
-router.get( "/", ( req, res ) => res.render( "welcome", { title: "wd-todo" } ) );
+router.get( "/", ( req, res ) => res.render( "landing" ) );
```
Please give the page a title, see line below for how.

jneidel commented:
Todo: create a favicon (see: https://github.com/jneidel/htw-webdev-todo/blob/master/public/pug/layout.pug#L7), can be easily done in GIMP

I would hold on with this until we have a set colorscheme though.

jneidel commented on public/pug/landing.pug:
```diff
@@ -0,0 +1,45 @@
+    link(rel="stylesheet" type="text/css" href="../css/landing.css")
+
+block main
+    section(class="container")
```
Use the shorthand .container

here `section.container`

jneidel commented on public/pug/landing.pug:
```diff
@@ -0,0 +1,45 @@
+                        p learn more
+                        img(src="/assets/img/tick.svg")
+
+block scripts
```
No need to import vue if you are not using it.

jneidel commented on public/css/landing.css:
```diff
@@ -0,0 +1,83 @@
+}
+  
+.btn{
+  background-color: rgb(180, 180, 180);
```
Can we use hex colors instead of rgb? They're easier to handle imo.

jneidel commented on public/css/landing.css:
```diff
@@ -0,0 +1,83 @@
+  bottom: -20px;
+}
+
+img{
```
Just add a .tick class to make it more percise.

Having this and adding another img at a later point is a great bug.

jneidel commented on public/css/landing.css:
```diff
@@ -0,0 +1,83 @@
+  margin-left: 1rem;
+}
+
+.btn:hover{
```
Put the hover, directly below the .btn styles to make it easier to read.

jneidel commented on public/css/landing.css:
```diff
@@ -0,0 +1,83 @@
+  align-items: center;
+}
+
+.center{
```
unused

jneidel commented on public/css/landing.css:
```diff
@@ -0,0 +1,83 @@
+  border-radius: .5rem;
+}
+
+.lists{
```
Make them ids (#) if only used once (and there is no potential for reuse).

jneidel reviewed (changes requested):
The boxes are cool and I like the colors individually, but with the white background it just doesn't come together for me.

Where are the "learn more" links going? Do we really need to "explain" these features in more depth?

What about a big version of the logo as the first thing of the page?
"Be more productive" doesn't really grab your attention.

You can color the header/footer as you see fit as well.

jneidel commented:
Once you are ready with the PR, you can deploy this branch to the heroku instace [here](https://dashboard.heroku.com/apps/htw-wd-todo/deploy/github) using `Manual deploy`.

enze-l committed:
Rework design to be more concise

- Moved logo out of header into body and changed size
- Removed introductional text which didn't have enough value
- Removed Buttons which did't have valuable information behind them

enze-l committed:
Resolve issues in layout

- colors are defined via hex-code now
- ids are now used for single occurences of a class

jneidel commented on routes/index.js:
```diff
@@ -1,7 +1,7 @@
 const router = express.Router();
 
-router.get( "/", ( req, res ) => res.render( "landing" ) );
+router.get( "/", ( req, res ) => res.render( "landing", { title: "landing-page"} ) );
```
No, more like "What to do?", the user does not care that its the landing page.

enze-l replied:
by changing the heading to "What to do", the tab-text will read like "What to do? | What to do?" which will just be redundand information. That is why I didn't give it an title to begin with. Would an tilte like "welcome" be better?

jneidel replied:
"What to do? | What to do?" is fine, though.

jneidel replied:
Again, it's not about the individual page, but the website as a whole. imo having a consistent title naming across all pages is more important than reducing some "redundant" info at points.

jneidel commented on public/pug/landing.pug:
```diff
@@ -4,42 +4,30 @@ block styles
-            "What to do" is an easy to use to-do application, designed for
-            individual or colaborativ use.
+    .container
+        p#logo.landing What to do?
```
Don't take the logo out the header. Don't touch the header and add another one to the landing page.

If you want to reuse the styles, put the relevant ones in a class, but don't steal the headers id.

enze-l replied:
By leaving the Logo in the heading, there will be two times the same logo next to itself, which would be redundant. Would't I be just copying and pasting those styles? I wanted to reuse them with this approach

jneidel replied:
> Would't I be just copying and pasting those styles

Put them in a class (that what I meant w/ "reuse the styles").

jneidel replied:
> By leaving the Logo in the heading, there will be two times the same logo next to itself, which would be redundant.

It's about how it looks, not it's information content. And like this it look horrible and super inconsistent, as all other site will have the logo in the header.
The big logo at the beginning of the page looks great, having a small version of it in the corner will not detract from it.

jneidel replied:
An alternative would be removing the header from the landing page. Having a link to the app/login site within the content.

jneidel commented on public/css/landing.css:
```diff
@@ -11,27 +11,21 @@ body {
-img{
-  height: 18px;
-  margin-left: 1rem;
+#logo{
```
Like I said in the html, don't mess with the headers styles.

jneidel commented on public/pug/landing.pug:
```diff
@@ -4,42 +4,30 @@ block styles
+                        Want to have seperate lists for work, home and shopping
+                        or a project is around the corner? Sort your tasks to be able 
+                        to focus on what matters.
+                    img.tick(src="/assets/img/tick.svg")
```
What function do the ticks serve now?

enze-l replied:
Both the color-coding and the ticks should compliment the text in representing the functionality of the app by not only describing the feature set but showing visually how task might be represented

jneidel reviewed (changes requested):
Still a lot left left to be desired. Didn't address all the point I mad within the last review.

It's not really my task, to keep reviewing every small change (or at all). So request a review once you find it to be completely ready. If you have questions I'm always here though, or you want to discuss colors, content, general structure, etc.
But critiquing along the way takes up too much time. And it also takes away from your agency, as you'll fall back on me making the decisions.

enze-l committed:
Add background-picture and do minor changes

- The "What to do?" logo now has its own class
- The title of the landing-page has changed to "What to do?"
- The color of the header has changed to white instead of transparent

enze-l committed:
Add forgoten picture

enze-l committed:
Make wallpaper of landing scale to screen-size

enze-l committed:
Merge branch 'master' into landing-page

enze-l committed:
Replace placeholder favicon and replace rgba with hexcodes

enze-l committed:
Merge branch 'landing-page' of https://github.com/jneidel/htw-webdev-todo into landing-page

enze-l committed:
Make major layout redesign

- Change logo font to VarelaRound
- Change header and footer layout
- Add blue-btn class
- Change Background to custom one
- Make custom favicon

enze-l committed:
Revert change in layout.css regarding grid-columns

enze-l committed:
Added tiling grid wallpaper

enze-l committed:
Change background

enze-l committed:
Make background scale with content

jneidel commented:
@zink-l  Wenn das Feature fertig ist kannst du es auch in eine normale PR umwandeln und den Prof zum abnehmen zuweisen.

viktorianer commented on public/pug/landing.pug:
```diff
@@ -0,0 +1,16 @@
+extends layout
```
Wie sorgt ihr dafür, dass eine Änderung an einem Template auch allen Clients bekannt gemacht wird? Jetzt ist es ja so, dass alles was im `public` liegt auch gecacht wird und damit die Änderungen nicht sofort sichtbar werden? Oder?

jneidel replied:
Meinen Sie mit Änderungen neue Commits oder andere an das view übergebene Daten?

viktorianer replied:
Weder noch. Wenn man diese Datei ändert und die Anwendung neu deployt, mit dieser Änderung, kann das jeder Benutzer auch sofort sehen?
Beispiel: `a.blue-btn` wird zu `a.red-btn`.

jneidel replied:
Wenn Sie das caching meinen: das liegt in diesem Fall in Herokus Händen (also caching zwischen vers. deployments).

Ansonsten weiß ich nicht worauf Sie hinaus wollen. Wenn ich das layout.pug template ändere dann wird die Änderung sofort angezeigt, da ja express aus den pug Dateien das html dynamisch generiert. Wir haben ja keine build pipeline die am Ende eine statische html Datei ausspuckt.

viktorianer replied:
Genau auf so eine `build pipeline` wollte ich hinaus. Die Information habe ich gebraucht bzw. wollte ich von euch hören, um es richtig bewerten zu können. Danke.

viktorianer reviewed (approved):
Einfaches, aber sauberes Feature. 👍

jneidel committed:
Remove redundant spacing & query params

jneidel commented on public/css/landing.css:
```diff
@@ -0,0 +1,38 @@
+}
+  
+#splashtext{
+  font-family: Varela;
```
Fallback?

jneidel commented on public/css/layout.css:
```diff
@@ -37,26 +36,21 @@ body {
+.logo {
   font-size: 40px;
-  font-family: Alfien, Lato, Arial;
+  font-family: Varela, Lato, Arial;
```
If you aren't using Lato remove it and it's import.

jneidel commented on public/pug/landing.pug:
```diff
@@ -0,0 +1,16 @@
+block main
+    .container
+        p#splashtext.
+            Forget your Notes
```
Why the spacing? The result isn't spaced and it isn't read like that.

jneidel commented on public/pug/landing.pug:
```diff
@@ -0,0 +1,16 @@
+        p#splashmessage.
+            Ideas change. Paper gets lost.
+            Tasks need to be done anyway.
+        a.blue-btn#big(href="sign up") sign up
```
`#big` is unused

enze-l committed:
Implement p-request feedback
